### PR TITLE
Remove remote spaces

### PR DESCRIPTION
**Pull request by nvinuesa**
This patch completely removes the concept of remote spaces in Juju. We only had this for binding offers on different models, but it is not useful since none of the providers support `ProviderSpaceInfo` and we have removed the `AreSpacesRoutable` on the providers. 

The cleanup includes the checks on the application offers API.

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [X] Comments saying why design decisions were made
- [X] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

Bootstrap and create a model:
```
juju bootstrap localhost c
juju add-model m1
```
deploy grafana, create a second model and deploy influx on it:
```
juju deploy grafana
juju add-model m2
juju deploy influxdb
```
offer the `grafana-source` endpoint of influxdb and integrate it with grafana:
```
juju offer influxdb:grafana-source
juju switch m1
juju integrate grafana admin/m2.influxdb
```
Also, try a CMR with a controller on 3.5 and another with this branch:
```
# Bootstrap a controller using juju 3.5 and deploy influxdb:
juju bootstrap localhost ctrl-src1
juju add-model model-src1
juju deploy influxdb
juju offer influxdb:grafana-source  

# Bootstrap a second controller using this branch (4.0) and deploy grafana:
juju bootstrap localhost ctrl-src2
juju add-model model-src2
juju deploy grafana

# Integration between grafana and influxdb should work:
juju integrate grafana ctrl-src1:admin/model-src1.influxdb

juju status --relations
Model       Controller  Cloud/Region         Version      SLA          Timestamp
model-src2  ctrl-src2   localhost/localhost  4.0-beta3.1  unsupported  12:09:18+01:00

SAAS      Status  Store      URL
influxdb  active  ctrl-src1  admin/model-src1.influxdb

App      Version  Status  Scale  Charm    Channel        Rev  Exposed  Message
grafana           active      1  grafana  latest/stable   69  no       Ready

Unit        Workload  Agent  Machine  Public address  Ports     Message
grafana/0*  active    idle   0        10.170.236.239  3000/tcp  Ready

Machine  State    Address         Inst id        Base          AZ  Message
0        started  10.170.236.239  juju-3f7181-0  ubuntu@20.04      Running

Integration provider     Requirer                Interface       Type     Message
influxdb:grafana-source  grafana:grafana-source  grafana-source  regular
```
## Links

**Jira card:** JUJU-5663

### CODE REVIEW

**Review by wallyworld — APPROVED** [review]

**Review by manadart — COMMENTED** [review]
I know you're doing the version change thing, but this patch could land against 3.5 if you just remove the changes to the RPC params (same compatibility, just never used).

Then do the version bumps separately.

**Comment by nvinuesa** [comment]
@manadart the problem is that we already merged the 3.5 to main and I need this patch for https://github.com/juju/juju/pull/17001. In 3.5 we ignore the spaces arguments since we weren't doing anything with them anyways

File: api/facadeversions.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ var facadeVersions = facades.FacadeVersions{
 	"AllWatcher":                   {3},
 	"Annotations":                  {2},
 	"Application":                  {19, 20},
-	"ApplicationOffers":            {4, 5},
```
Comment (SimonRichardson) [inline api/facadeversions.go:33]: Just a mental note for me, we're dropping support for which version of Juju here (2.x or 3.x)?

Comment (nvinuesa) [reply api/facadeversions.go:33]: this is 3.x

Comment (SimonRichardson) [reply api/facadeversions.go:33]: We need to make sure we don't leave 3.x orphaned here.

**Comment by SimonRichardson** [comment]
@nvinuesa can you do a controller CMR to a 3.4/3.5 to ensure it works.

**Comment by nvinuesa** [comment]
@SimonRichardson updated the QA steps with the CMR 3.5->4.0

**Pull request merged**